crates/red_knot_python_semantic/src/node_key.rs

```diff
@@ -17,8 +23,15 @@ impl NodeKey {
     {
         let node = node.into();
         NodeKey {
-            kind: node.kind(),
+            kind: Kind::Node(node.kind()),
             range: node.range(),
         }
     }
+
+    pub(super) fn from_identifier(identifier: &Identifier) -> Self {
+        NodeKey {
+            kind: Kind::Identifier,
+            range: identifier.range(),
+        }
+    }
 }
```

Comment:
There's not much in it, but I'd be tempted to do something like this (diff is relative to your branch):

```diff
diff --git a/crates/red_knot_python_semantic/src/node_key.rs b/crates/red_knot_python_semantic/src/node_key.rs
index 9683b0e7f..6aead4ce6 100644
--- a/crates/red_knot_python_semantic/src/node_key.rs
+++ b/crates/red_knot_python_semantic/src/node_key.rs
@@ -21,14 +21,21 @@ impl NodeKey {
     where
         N: Into<AnyNodeRef<'a>>,
     {
-        let node = node.into();
+        Self::from(node.into())
+    }
+}
+
+impl From<AnyNodeRef<'_>> for NodeKey {
+    fn from(node: AnyNodeRef) -> Self {
         NodeKey {
             kind: Kind::Node(node.kind()),
             range: node.range(),
         }
     }
+}
 
-    pub(super) fn from_identifier(identifier: &Identifier) -> Self {
+impl From<&Identifier> for NodeKey {
+    fn from(identifier: &Identifier) -> Self {
         NodeKey {
             kind: Kind::Identifier,
             range: identifier.range(),
diff --git a/crates/red_knot_python_semantic/src/semantic_index/definition.rs b/crates/red_knot_python_semantic/src/semantic_index/definition.rs
index 75c95a4bd..a54dd67fc 100644
--- a/crates/red_knot_python_semantic/src/semantic_index/definition.rs
+++ b/crates/red_knot_python_semantic/src/semantic_index/definition.rs
@@ -462,6 +462,6 @@ impl From<&ast::ParameterWithDefault> for DefinitionNodeKey {
 
 impl From<&ast::Identifier> for DefinitionNodeKey {
     fn from(identifier: &ast::Identifier) -> Self {
-        Self(NodeKey::from_identifier(identifier))
+        Self(NodeKey::from(identifier))
     }
 }
```